modules/case-management/src/test/java/com/ibm/cloud/platform_services/case_management/v1/CaseManagementIT.java

```diff
@@ -0,0 +1,367 @@
+            .offset(Long.valueOf("0"))
+            .limit(Long.valueOf("2"))
+            .sort("number")
+            .status(new ArrayList<String>(Arrays.asList("new")))
+            .fields(new ArrayList<String>(Arrays.asList(fields)))
+            .build();
+
+        Response<CaseList> response = service.getCases(getCasesOptionsModel).execute();
+        assertNotNull(response);
+        CaseList responseObj = response.getResult();
+
+        assertTrue(0 < responseObj.getTotalCount());
+        assertTrue(0 < responseObj.getCases().size());
+    }
+
+    @Test (dependsOnMethods = {"testCreateCase"})
+    public void testGetCase() {
+
+        String[] fields = new String[] {"number", "description"};
+
+        GetCaseOptions getCaseOptionsModel = new GetCaseOptions.Builder()
+            .caseNumber(newCaseNumber)
+            .fields(new ArrayList<String>(Arrays.asList(fields)))
+            .build();
+
+        Response<Case> response = service.getCase(getCaseOptionsModel).execute();
+        assertNotNull(response);
+        Case responseObj = response.getResult();
+        assertEquals(newCaseNumber, responseObj.getNumber());
+    }
+
+    @Test (dependsOnMethods = {"testCreateCase"}, expectedExceptions = {BadRequestException.class})
+    public void testGetCasewithInvalidField() {
+
+        String[] fields = new String[] {"number", "description", "invalid_field"};
+
+        GetCaseOptions getCaseOptionsModel = new GetCaseOptions.Builder()
+            .caseNumber(newCaseNumber)
+            .fields(new ArrayList<String>(Arrays.asList(fields)))
+            .build();
+
+        Response<Case> response = service.getCase(getCaseOptionsModel).execute();
+        assertNotNull(response);
+        Case responseObj = response.getResult();
+        assertEquals(newCaseNumber, responseObj.getNumber());
+    }
+
+    @Test (dependsOnMethods = {"testCreateCase"})
+    public void testAddComment() {
+
+        String comment = "This is a test comment!";
+
+        AddCommentOptions addCommentOptionsModel = new AddCommentOptions.Builder()
+            .caseNumber(newCaseNumber)
+            .comment(comment)
+            .build();
+
+        Response<Comment> response = service.addComment(addCommentOptionsModel).execute();
+
+        Comment responseObj = response.getResult();
+        assertEquals(comment, responseObj.getValue());
+    }
+
+    @Test (expectedExceptions = {NotFoundException.class})
+    public void testAddCommentToNonExistingCase() {
+
+        String comment = "This is a test comment!";
+
+        AddCommentOptions addCommentOptionsModel = new AddCommentOptions.Builder()
+            .caseNumber("uncorn-case-num")
+            .comment(comment)
+            .build();
+
+        Response<Comment> response = service.addComment(addCommentOptionsModel).execute();
+
+        Comment responseObj = response.getResult();
+        assertEquals(comment, responseObj.getValue());
```

Comment:
`responseObj` will be an error object. Can't assert it to be equal the `comment`.

Reply:
Good catch. I will remove line 251 and 252 as they are not reached.

Reply:
Made necessary changes here and all other places.